### PR TITLE
Bug fix for https://bugs.launchpad.net/or/+bug/1895692 False done indication on drop cars at location; by edwardk

### PR DESCRIPTION
**Pull request by Csantucci**
See also http://www.elvastower.com/forums/index.php?/topic/34402-issue-with-dropoffwagonsatlocation-event/page__pid__262879#entry262879

### CODE REVIEW

File: Source/Orts.Simulation/Simulation/Activity.cs [hunk 1]
```diff
@@ -1334,7 +1334,7 @@ override public Boolean Triggered(Activity activity)
                     {
                         consistTrain = null;
```
Comment (YoRyan) [inline Source/Orts.Simulation/Simulation/Activity.cs:1335]: Side note, can't we delete this line? It's a redundant assignment.

File: Source/Orts.Simulation/Simulation/Activity.cs [hunk 2]
```diff
@@ -1334,7 +1334,7 @@ override public Boolean Triggered(Activity activity)
...
                         consistTrain = matchesConsistNoOrder(ChangeWagonIdList);
-                        triggered = (consistTrain != null ? false : true);
+                        triggered = (consistTrain != null ? true : false);
```
Comment (YoRyan) [inline Source/Orts.Simulation/Simulation/Activity.cs:1337]: Can be simplified to:
```suggestion
                        triggered = consistTrain != null;
```

File: Source/Orts.Simulation/Simulation/Activity.cs [hunk 3]
```diff
@@ -1388,12 +1388,11 @@ private Train matchesConsistNoOrder(List<string> wagonIdList)
         {
             foreach (var trainItem in Simulator.Trains)
             {
-                bool lEngine = false;
+                //bool lEngine = false;
```
Comment (YoRyan) [inline Source/Orts.Simulation/Simulation/Activity.cs:1391]: It's bad form to litter the source code with commented code sections. Just delete it.
```suggestion
```

File: Source/Orts.Simulation/Simulation/Activity.cs [hunk 4]
```diff
@@ -1438,7 +1435,19 @@ static bool includesWagons(Train train, List<string> wagonIdList)
...
+            {
+                var car = train.Cars[i];
+                if (car.Train.TrainType == Train.TRAINTYPE.STATIC)
+                {
+                    lNotFound = true;
+                    return lNotFound;
+                }
+
+            }
+
```
Comment (YoRyan) [inline Source/Orts.Simulation/Simulation/Activity.cs:1450]: Correct me if I'm wrong, but couldn't this simplify to:
```suggestion
            // The Cars list is a global list that includes STATIC cars.  We need to make sure that the active train/car is processed only.
            if (train.TrainType == Train.TRAINTYPE.STATIC)
                return true;

            bool lNotFound = false;
```

File: Source/Orts.Simulation/Simulation/Activity.cs (hunk 1 above)
Comment (Csantucci) [reply Source/Orts.Simulation/Simulation/Activity.cs]: OK, line removed together with a similar nearby one.

**Review by YoRyan — APPROVED** [review]
Thanks, Carlo. This change now looks good to me.

**Pull request merged**